test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/RuntimeTypeTestsJSTest.scala

```diff
@@ -0,0 +1,163 @@
+    testArraySomeJSClass(true, arrayOfJSClass)
+
+    testArrayParentJSType(true, arrayOfJSInterface)
+    testArrayParentJSType(true, arrayOfJSClass)
+
+    testArraySomeJSInterface(false, arrayOfParentJSType)
+    testArraySomeJSClass(false, arrayOfParentJSType)
+
+    testArrayJSObject(false, arrayOfJSInterface)
+    testArrayJSObject(true, arrayOfJSClass)
+  }
+
+  @Test def objectType(): Unit = {
+    @inline def testObject(expected: Boolean, value: Any): Unit =
+      test(expected, value, classOf[Object], _.isInstanceOf[Object], _.asInstanceOf[Object])
+
+    testObject(true, new MyJSClass)
+    testObject(true, new MyJSSubClass)
+    testObject(true, new js.Object)
+  }
+
+  @Test def jsObject(): Unit = {
+    @inline def testJSObject(expected: Boolean, value: Any): Unit =
+      testJS(expected, value, classOf[js.Object], _.isInstanceOf[js.Object], _.asInstanceOf[js.Object])
+
+    testJSObject(true, new MyJSClass)
+    testJSObject(true, new MyJSSubClass)
+    testJSObject(true, new js.Object)
+    testJSObject(true, List(5))
+    testJSObject(true, new Exception)
+
+    // characters are always boxed
+    testJSObject(true, 'A')
+
+    // longs are boxed iff they are not bigints
+    val longsAreBigInts = js.typeOf(5L) == "bigint"
+    testJSObject(!longsAreBigInts, 5L)
+
+    testJSObject(false, 5)
+    testJSObject(false, ())
+    testJSObject(false, true)
+
+    testNullValue(classOf[js.Object], _.isInstanceOf[js.Object], _.asInstanceOf[js.Object])
+  }
+
+  @Test def jsError(): Unit = {
+    @inline def testJSError(expected: Boolean, value: Any): Unit =
+      testJS(expected, value, classOf[js.Error], _.isInstanceOf[js.Error], _.asInstanceOf[js.Error])
+
+    testJSError(true, new js.Error)
+    testJSError(true, new js.EvalError)
+    testJSError(true, new Exception)
+
+    testJSError(false, 5)
+    testJSError(false, ())
+    testJSError(false, true)
+
+    testNullValue(classOf[js.Error], _.isInstanceOf[js.Error], _.asInstanceOf[js.Error])
+  }
+
+  @Test def jsClass(): Unit = {
+    @inline def testMyJSClass(expected: Boolean, value: Any): Unit =
+      testJS(expected, value, classOf[MyJSClass], _.isInstanceOf[MyJSClass], _.asInstanceOf[MyJSClass])
+
+    testMyJSClass(true, new MyJSClass)
+    testMyJSClass(true, new MyJSSubClass)
+
+    testMyJSClass(false, new js.Object)
+    testMyJSClass(false, List(5))
+
+    testNullValue(classOf[MyJSClass], _.isInstanceOf[MyJSClass], _.asInstanceOf[MyJSClass])
+  }
+
+  @Test def jsTrait(): Unit = {
+    assertThrows(classOf[Exception], classOf[MyJSTrait].isInstance(5))
+    assertThrows(classOf[Exception], classOf[MyJSTrait].isInstance(new MyJSClass))
+    assertThrows(classOf[Exception], classOf[MyJSTrait].isInstance(null))
+  }
+
+}
+
+object RuntimeTypeTestsJSTest {
+
+  @js.native
+  trait ParentJSType extends js.Object
+
+  @js.native
+  trait SomeJSInterface extends ParentJSType
+
+  @JSGlobal("SomeJSClass")
+  @js.native
+  class SomeJSClass extends ParentJSType
```

Comment:
I do not understand the rationale of when native / non-native types are used here.

Reply:
There was no rationale. It was an artifact of merging tests from two different places. I have replaced with only using non-native types.